catalog/task/cleanup-workspace/0.2/tests/test-cleanup-workspace.yaml

```diff
@@ -0,0 +1,62 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: test-cleanup-workspace
+spec:
+  description: |
+    Run the cleanup-workspace task with a custom workspace
+    and verify that the cleanup is performed in the given subdirectory.
+  workspaces:
+    - name: tests-workspace
+  tasks:
+    - name: setup
+      workspaces:
+        - name: input
+          workspace: tests-workspace
+      taskSpec:
+        workspaces:
+          - name: input
+        steps:
+          - name: put-content-in-workspace
+            image: quay.io/hacbs-release/release-utils:5b1a1cd9fd68625cab5573ce62e0d87e6f93f341
+            script: |
+              #!/usr/bin/env sh
+              
+              INPUT_PATH="$(workspaces.input.path)"
+              mkdir -p $INPUT_PATH/dir-to-be-cleaned-up/
+              echo "This file and dir will be deleted" >> $INPUT_PATH/dir-to-be-cleaned-up/cleanup_test.txt
+              mkdir -p $INPUT_PATH/dir-to-be-kept/
+              echo "Test successful" >> $INPUT_PATH/dir-to-be-kept/cleanup_check.txt
+              ls -R $INPUT_PATH/
+    - name: run-task
+      taskRef:
+        name: cleanup-workspace
+      params:
+        - name: subdirectory
+          value: "dir-to-be-cleaned-up"
+      runAfter:
+        - setup
+      workspaces:
+        - name: input
+          workspace: tests-workspace
+    - name: check-result
+      workspaces:
+        - name: input
+          workspace: tests-workspace
+      runAfter:
+        - run-task
+      taskSpec:
+        workspaces:
+          - name: input
+        steps:
+          - name: check-result
+            image: quay.io/hacbs-release/release-utils:5b1a1cd9fd68625cab5573ce62e0d87e6f93f341
+            script: |
+              #!/usr/bin/env sh
+
+              INPUT_PATH="$(workspaces.input.path)"
```

Comment:
This seems a bit odd that one subdir has a dedicated variable and the other not. I would suggest just creating the two (and no `INPUT_PATH`):
```
REMOVED_DIR="$(workspaces.input.path)/dir-to-be-cleaned-up"
KEPT_DIR="$(workspaces.input.path)/dir-to-be-kept"
```